src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -868,6 +901,19 @@ internal bool TryGetCollectionIterationType(ExpressionSyntax syntax, TypeSymbol
                 var elementType = elementTypeWithAnnotations.Type;
                 Debug.Assert(elementType is { });
 
+                if (collectionTypeKind == CollectionExpressionTypeKind.ImplementsIEnumerable)
+                {
+                    if (!HasCollectionExpressionApplicableConstructor(node.Syntax, targetType, diagnostics))
+                    {
+                        reportedErrors = true;
```

Comment:
>reportedErrors = true; [](http://example.com/codeflow?start=24&length=22)

Are we covering this code path for a type parameter?

Reply:
Yes, there are several tests that hit this with a type parameter, including `TypeParameter_03`.